course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,216 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# An Introduction to Dask\n",
+    "\n",
+    "Dask is a flexible parallel computing library for Python. It allows dynamic scheduling of tasks for optimising computations, and includes collections such as arrays and lists for parallel data processing."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Terminology\n",
+    "\n",
+    "### General"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Thread\n",
+    "A thread is the simplest unit of computation.\n",
+    "\n",
+    "#### Process\n",
+    "A process is an isolated computation which consists of one or more threads."
```

Comment:
Both of these definitions seem a little terse! Could we perhaps put a little more meat on the bones that connects the (excellent) definitions to the context we're working in?

Reply:
How about:

#### Thread
A thread is the simplest unit of computation.

#### Process
A process is an isolated computation which consists of one or more threads. The threads within a process can be executed concurrently (simultaneously) and have access to the same resources within the process (memory, executable code and variable values). Processes cannot share these resources with other processes. Each process has its own individual address space, while the threads share their address space.